### ISSUE TITLE
`typing.overload` by any other name (or source) confuses Flake8 (but Mypy gets it right)

### ISSUE THREAD

**Issue by posita** [PyCQA/flake8#1575]
### how did you install flake8?

```console
% pip install flake8
```

### unmodified output of `flake8 --bug-report`

```json
{
  "dependencies": [],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.9.10",
    "system": "Darwin"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.8.0"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "2.4.0"
    }
  ],
  "version": "4.0.1"
}
```

### describe the problem

[Beartype](https://github.com/beartype/beartype/) tries to unifies typing disparities across Python versions, and (re)exports `typing` primitives in `beartype.typing`. This provides a convenience that users can do `from beartype.typing import …` and not have to worry about whether that comes from `typing_extensions` on version 3.X or from `typing` on 3.Y, etc. The problem is that `from beartype.typing import overload` (which is [just a reexport of `typing.overload`](https://github.com/beartype/beartype/blob/76b005e088875277f11d554d61df7f71427794f5/beartype/typing/__init__.py#L138)) confuses Flake8. Consider:

``` python
# test_case.py
from beartype.typing import Optional, Union, overload

# from typing import overload  # <-- importing overload this way will fix things


@overload
def foo() -> None:
    ...


@overload
def foo(arg: str) -> str:
    ...


@overload
def foo(arg: int, other_arg: Optional[int]) -> float:
    ...


def foo(
    arg: Optional[Union[int, str]] = None, other_arg: Optional[int] = None
) -> Optional[Union[float, str]]:
    if arg is None or isinstance(arg, str):
        return arg
    elif isinstance(arg, int) and isinstance(other_arg, int):
        return arg / other_arg
    else:
        assert False, f"unknown type(s) for {arg} and {other_arg}"
```

``` sh
% flake8 test_case.py
test_case.py:13:1: F811 redefinition of unused 'foo' from line 8
test_case.py:18:1: F811 redefinition of unused 'foo' from line 13
test_case.py:22:1: F811 redefinition of unused 'foo' from line 18
```

Those errors go away when importing `typing.overload` directly.

Note, however, Mypy understands:

``` sh
% mypy --version
mypy 0.941
% mypy --config-file=/dev/null test_case.py
/dev/null: No [mypy] section in config file
Success: no issues found in 1 source file
```

**Comment by posita** [comment]
@asottile, this is [the second issue of mine](https://github.com/pre-commit/identify/issues/289) that you have closed within minutes of me filing it and without any explanation. Can you offer some insight as to why this should be closed?

**Comment by asottile** [comment]
you didn't follow the issue form or read the issue form which addresses your question

**Comment by posita** [comment]
Thank you.

**Comment by posita** [comment]
Can you please reopen this and transfer it to [PyCQA/pyflakes](https://github.com/PyCQA/pyflakes)? I'm pretty sure it belongs there. pyflakes#626, pyflakes#592, and pyflakes#482 are all closed as either dupes or as fixed. This is either a new issue, or it's a regression. Thanks!

**Comment by posita** [comment]
Is that true? PyCQA/pyflakes#632 (which fixes PyCQA/pyflakes#561) does do some inspection regarding the source of `overload`. I can look into this, but if I present a PR in the pyflakes project, are you just going to close it without looking at it?
